From c7126601874a8b10dd13108ba7569e3c85c6d4a6 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Fri, 9 Jun 2023 11:28:47 +0800
Subject: [PATCH 404/780] [Huawei] block: fix kabi broken in struct bio

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188826

--------------------------------

Use kabi reserved field for new member in struct bio.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 include/linux/blk_types.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index d4030b3bba40..fdb75fd6a38f 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -264,10 +264,6 @@ struct bio {
 
 	struct bio_set		*bi_pool;
 
-#ifdef CONFIG_BLK_IO_HIERARCHY_STATS
-	u64			hierarchy_time;
-#endif
-
 	/*
 	 * Use the first RESERVE field for EulerOS
 	 * scsi log enhance feature
@@ -278,7 +274,11 @@ struct bio {
 #else
 	KABI_RESERVE(2)
 #endif
+#ifdef CONFIG_BLK_IO_HIERARCHY_STATS
+	KABI_USE(3, u64 hierarchy_time)
+#else
 	KABI_RESERVE(3)
+#endif
 	KABI_RESERVE(4)
 
 	/*
-- 
2.17.1

